docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+```
+
+Compare this with accessing the vector contained in column `3`:
+
+```jldoctest dataframe
+julia> german[:, 3]
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+We can get a single cell or element with the same syntax to get an element of an array:
+```jldoctest dataframe
+julia> german[4, 4]
+2
+```
+
+or to get a new `DataFrame` that is subset of rows and columns:
+```jldoctest dataframe
+julia> german[4:5, 4:5]
+2×2 DataFrame
+ Row │ Job    Housing
+     │ Int64  String
+─────┼────────────────
+   1 │     2  free
+   2 │     2  free
+```
+
+Assignment of a scalar to a data frame can be done in ranges using broadcasting:
+```jldoctest dataframe
+julia> german[:, 2] .= 4 # an in-place replacement of values stored in column number 2 by 4
+1000-element view(::Vector{Int64}, :) with eltype Int64:
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ ⋮
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+
+julia> german
+1000×10 DataFrame
+  Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking accoun ⋯
+      │ Int64  Int64  String  Int64  String   String           String          ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0      4  male        2  own      NA               little          ⋯
+    2 │     1      4  female      2  own      little           moderate
+    3 │     2      4  male        1  own      little           NA
+    4 │     3      4  male        2  free     little           little
+    5 │     4      4  male        2  free     little           little          ⋯
+    6 │     5      4  male        1  free     NA               NA
+    7 │     6      4  male        2  own      quite rich       NA
+    8 │     7      4  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮      ⋮       ⋮       ⋮            ⋮                ⋮        ⋱
+  994 │   993      4  male        3  own      little           little          ⋯
+  995 │   994      4  male        2  own      NA               NA
+  996 │   995      4  female      1  own      little           NA
+  997 │   996      4  male        3  own      little           little
+  998 │   997      4  male        2  own      little           NA              ⋯
+  999 │   998      4  male        2  free     little           little
+ 1000 │   999      4  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+```
+
+Using `setindex!` on *DataFrame* :  
+
+```jldoctest dataframe
+julia> data = DataFrame(marks = repeat(70:72, inner = 2),
```

Comment:
I think we should stick to `german` data frame in all examples. What is the benefit of introducing a new object?

Reply:
The main reason behind the introduction of a new object is that `german` data frame has around 1000 rows and 10 columns so it's been hard to explain some examples because it omits the output after compiling and then because of it we lost the purpose of those concepts. So, it's better to use a new object to explain those concepts and in the end, our main purpose is to explain to users that how we should use these functions in our work. So, it completes our purpose. And it's our introductory tutorial and later in the `gallery` examples, we can make a complete tutorial to do the analysis on some data frames.

Reply:
Why not make a smaller subset of a `german` data frame and work on it then? (this was one of the objectives of the creation of the tutorial as we discussed when we started the project - using one data set consistently in the tutorial was requested by the users, as in current documentation we create ad-hoc data frames and users were confused by switching of the context)

If you strictly prefer to stick to the other data frame example please add some explanation why we do it in the material (and that it is a temporary thing; still - I would prefer to stick to a single data set example if possible)